robocop/checkers/naming.py

```diff
@@ -263,43 +269,83 @@
+        ),
+        rule_id="0320",
+        name="not-allowed-char-in-filename",
+        msg="Not allowed character '{{ character }}' found in {{ block_name }} name",
+        severity=RuleSeverity.WARNING,
+        docs="""
+        Reports not allowed pattern found in Suite names. By default it's dot (`.`). You can
+        configure what characters are reported by calling::
+
+             robocop --configure not-allowed-char-in-filename:pattern:regex_pattern
+
+        `regex_pattern` should define regex pattern for characters not allowed in names. For example `[@\[]` pattern
+        reports any occurrence of `@[` characters.
+        """,
+    ),
 }
 
 
-class InvalidCharactersInNameChecker(VisitorChecker):
+class InvalidCharactersInFileNameChecker(VisitorChecker):
     """Checker for invalid characters in suite, test case or keyword name."""
 
-    reports = ("not-allowed-char-in-name",)
+    reports = ("not-allowed-char-in-filename",)
 
     def visit_File(self, node):
         source = node.source if node.source else self.source
         if source:
             suite_name = Path(source).stem
             if "__init__" in suite_name:
                 suite_name = Path(source).parent.name
-            for char in suite_name:
-                if self.param("not-allowed-char-in-name", "pattern").search(char):
-                    self.report("not-allowed-char-in-name", character=char, block_name="suite", node=node)
+            for iter in self.param("not-allowed-char-in-filename", "pattern").finditer(suite_name):
+                self.report(
+                    "not-allowed-char-in-filename",
+                    character=iter.group(),
+                    block_name="suite",
+                    node=node,
+                    col=node.col_offset + iter.start(0) + 1,
+                )
         super().visit_File(node)
 
+
+class InvalidCharactersInNameChecker(VisitorChecker):
+    """Checker for invalid characters in suite, test case or keyword name."""
+
+    reports = ("not-allowed-char-in-name",)
+
     def check_if_char_in_node_name(self, node, name_of_node, is_keyword=False):
-        variables = find_robot_vars(node.name) if is_keyword else []
-        index = 0
-        while index < len(node.name):
-            # skip variables
-            if variables and variables[0][0] == index:
-                start, stop = variables.pop(0)
-                index += stop - start
-            else:
-                if self.param("not-allowed-char-in-name", "pattern").search(node.name[index]):
-                    self.report(
-                        "not-allowed-char-in-name",
-                        character=node.name[index],
-                        block_name=f"'{node.name}' {name_of_node}",
-                        node=node,
-                        col=node.col_offset + index + 1,
-                    )
-                index += 1
+        node_name = node.name
```

Comment:
I still vividly remember I had to introduce some hacks here to handle variables, patterns and still be able to pinpoint location of the issue. I now need to checkout your branch and debug that part a bit to fully get it :)

Reply:
I'll add some comments to code.
Your code was looping `node.name` char by char and dropping variables off if there was one. I had to change variables searching, because in my proposal will search pattern from fully string.